addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanBaseSensorHandler.java

```diff
@@ -153,36 +163,41 @@ public void espPacketReceived(ESP3Packet packet) {
-
-                                    // if message can be interpreted (result != UnDefType.UNDEF) => update item state
-                                    if (result != null && result != UnDefType.UNDEF) {
-                                        updateState(channelId, result);
-                                        setCurrentState(channelTypeId, result); // update internal state map
-                                    }
-                                    break;
-                                case TRIGGER:
-                                    String lastEvent = lastEvents.get(channelId);
-                                    String event = eep.convertToEvent(channelId, channelTypeId, lastEvent,
-                                            channelConfig);
-                                    if (event != null) {
-                                        triggerChannel(channel.getUID(), event);
-                                        lastEvents.put(channelId, event);
-                                    }
-                                    break;
-                            }
-                        });
+                streamSupplier.get().filter(stateFilter(receivingEEPType, senderId)).forEach(channel -> {
+                    ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
+
+                    String channelTypeId = (channelTypeUID != null) ? channelTypeUID.getId() : "";
+                    String channelId = channel.getUID().getId();
+                    Configuration channelConfig = channel.getConfiguration();
+
+                    State currentState = getCurrentState(channelId);
+                    State result = eep.convertToState(channelId, channelTypeId, channelConfig, currentState);
+
+                    // if message can be interpreted (result != UnDefType.UNDEF) => update item state
+                    if (result != null && result != UnDefType.UNDEF) {
+                        updateState(channelId, result);
+                        setCurrentState(channelTypeId, result); // update internal state map
+                    }
+                });
+
+                streamSupplier.get().filter(triggerFilter(receivingEEPType, senderId)).forEach(channel -> {
+                    ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
+
+                    String channelTypeId = (channelTypeUID != null) ? channelTypeUID.getId() : "";
+                    String channelId = channel.getUID().getId();
+                    Configuration channelConfig = channel.getConfiguration();
+
+                    String lastEvent = lastEvents.get(channelId);
+                    String event = eep.convertToEvent(channelId, channelTypeId, lastEvent, channelConfig);
+                    if (event != null) {
+                        triggerChannel(channel.getUID(), event);
+                        lastEvents.put(channelId, event);
+                    }
+                });
             }
         } catch (Exception e) {
```

Comment:
Could you please change that exception to more specific ones?

Reply:
Sure, thanks!